src/feature/dirauth/process_descs.c

```diff
@@ -241,10 +304,18 @@ dirserv_router_get_status(const routerinfo_t *router, const char **msg,
     return FP_REJECT;
   }
 
-  /* Check for the more common reasons to reject a router first. */
-  const uint32_t r = dirserv_get_status_impl(d, router->nickname,
-                                             router->addr, router->or_port,
-                                             router->platform, msg, severity);
+  /* First, check for the more common reasons to reject a router. */
+  if (router->cache_info.signing_key_cert) {
+    /* This has an ed25519 identity key. */
+    r = dirserv_get_status_impl(d,
```

Comment:
These two calls to dirserv_get_status_impl are almost identical.  Instead of having two calls like this, it's cleaner to have a local variable that holds either the ed25519 public key or NULL, and then pass that variable to a single call.